lib/cisco_node_utils/span_session.rb

```diff
@@ -0,0 +1,125 @@
+# Insert appropriate license here
+require_relative 'node_util'
+require_relative 'interface'
+
+
+module Cisco
+  # node utils class for SPAN (switchport analyzer) sessions
+  class SpanSession < NodeUtil
+    attr_reader :session_id
+
+    def initialize(session_id)
```

Comment:
Couple of things for the `initialize` method.

* Add an argument called `instantiate` and set it to `true` by default.
* Call your `create` method when `instantiate` is true so that the monitor session gets created.
* When you add the puppet layer, you will call this method with `instantiate` to false when you query for an existing list of `span_session` resources that are already configured.